guides/19_Video_Codecs.md

```diff
@@ -0,0 +1,71 @@
+
+The Amazon Chime SDK for Javascript supports setting a list of video codec preferences which will be used to select the encoder for an individual client. The API allows setting fallback codecs to, if desired, ensure that all participants in the call can receive video from senders as long as there is some intersection between the chosen codecs and the receive capabilities. This document will cover how to use the API, its effects, and some possible use cases for using it.
+
+## Setting Video Send Codec Preferences
+
+Application builders can use [`AudioVideoControllerFacade.setVideoCodecSendPreferences`](https://aws.github.io/amazon-chime-sdk-js/interfaces/audiovideocontrollerfacade.html#setvideocodecsendpreferences) for control over video send codecs. The fallback behavior is indicated in the documentation and can be made clear with a few examples. Not calling this function is functionally equivalent to the following:
+```
+// Assumes a meeting session has already been created and stored in `this.meetingSession`
+
+this.meetingSession.audioVideo.setVideoCodecSendPreferences([VideoCodecCapability.h264()]);
+```
+
+This implies the following:
+* Ignoring other restrictions, the sending client will transmit video into the meeting using the H.264 video codec.
+* The sending client *will not* fallback to any other codec if another attendee joins without an H.264 decoder (which may occur on some browsers, see notes below). That receiving attendee will not receive this video.
+* If the sending client does not support encoding H.264, then it will fall back to any video codec it supports, that is also supported by the meeting. Currently that would be VP8 only.
+
+For more complex fallback behavior, consider the following example:
+```
+this.meetingSession.audioVideo.setVideoCodecSendPreferences([VideoCodecCapability.h264(), VideoCodecCapability.vp8()]);
+```
+
+This implies the following:
+* Ignoring other restrictions, the sending client will transmit video into the meeting using the H.264 video codec.
+* The sending client *will* fallback to VP8 other codec if another attendee joins with H.264 decode support, but does have VP8 decode support.
+* If the sending client does not support encoding H.264, then it will fall back to VP8 (not technically a change in behavior, but will be more relevant when accounting for additional codecs).
+
+A similar API [`ContentShareControllerFacade.setContentShareVideoCodecPreferences`](https://aws.github.io/amazon-chime-sdk-js/interfaces/contentsharecontrollerfacade.html#setcontentsharevideocodecpreferences) exists to control content share video send codecs and it is used in identical fashion, e.g.:
```

Comment:
It is extremely relevant even with just VP8 and H.264. For example safari screen share has issue with H.264 encoder at high resolution, and additionally technically the only 'safe' way to do >720p is with VP8 (H.264 technically only requires decoders to support 720p max). This is more critical with future codecs, some which are better at screen share then others :).

Reply:
Ok, but is the 90% time ppl will just expect the same codec for both? If so, we can make content share to copy the setting from the sharer and they can override it if they want :) If you think most of the time ppl will set them up differently between video and content share then its fine :D

Reply:
I do not think it would be 90%. Given the possible large impact on CPU/video quality I would like builders to make each choice explicitly. Again the performance/quality impact of different codecs varies **significantly** between screen content and camera content. This is more of an advanced API anyways; I just don't want to make builders shoot themselves in the foot just so I can avoid having them call 2 functions.